File: app/src/main/java/com/example/earthwallet/ui/pages/staking/StakingTabsAdapter.java
package com.example.earthwallet.ui.pages.staking;

import android.util.Log;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

/**
 * ViewPager2 adapter for staking management tabs
 * Based on LiquidityTabsAdapter for consistent styling
 */
public class StakingTabsAdapter extends FragmentStateAdapter {

    private static final String TAG = "StakingTabsAdapter";

    public StakingTabsAdapter(Fragment fragment) {
        super(fragment);
        Log.d(TAG, "StakingTabsAdapter created");
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Log.d(TAG, "createFragment called for position: " + position);
        switch (position) {
            case 0: 
                Log.d(TAG, "Creating StakingInfoFragment");
                return StakingInfoFragment.newInstance();
            case 1: 
                Log.d(TAG, "Creating StakeUnstakeFragment");
                return StakeUnstakeFragment.newInstance();
            case 2: 
                Log.d(TAG, "Creating UnbondingFragment");
                return UnbondingFragment.newInstance();
            default: 
                Log.d(TAG, "Creating default StakingInfoFragment");
                return StakingInfoFragment.newInstance();
        }
    }

    @Override
    public int getItemCount() {
        return 3; // Info & Rewards, Stake/Unstake, Unbonding
    }
}

File: app/src/main/java/com/example/earthwallet/ui/pages/staking/StakingTabsAdapter.kt
package network.erth.wallet.ui.pages.staking

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter

/**
 * ViewPager2 adapter for staking management tabs
 * 5 tabs: Rewards, Stake, Unstake, Redelegate, Unbonding
 */
class StakingTabsAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {

    companion object {
        const val TAB_REWARDS = 0
        const val TAB_STAKE = 1
        const val TAB_UNSTAKE = 2
        const val TAB_REDELEGATE = 3
        const val TAB_UNBONDING = 4
    }

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            TAB_REWARDS -> RewardsFragment.newInstance()
            TAB_STAKE -> StakeFragment.newInstance()
            TAB_UNSTAKE -> UnstakeFragment.newInstance()
            TAB_REDELEGATE -> RedelegateFragment.newInstance()
            TAB_UNBONDING -> UnbondingFragment.newInstance()
            else -> RewardsFragment.newInstance()
        }
    }

    override fun getItemCount(): Int = 5
}
